docs/references/config/config.toml.md

```diff
@@ -0,0 +1,913 @@
+
+If [`grpc.laddr`](#grpcladdr) is empty, this setting is ignored and the service is not enabled.
+
+## Peer-to-peer
+These configuration options change the behaviour of the peer-to-peer protocol.
+
+### p2p.laddr
+TCP socket address for the P2P service to listen on.
+```toml
+laddr = "tcp://0.0.0.0:26656"
+```
+
+| Value type          | string                                            |
+|:--------------------|:--------------------------------------------------|
+| **Possible values** | TCP Stream socket (`"tcp://127.0.0.1:26657"`)     |
+
+### p2p.external_address
+TCP address to use as identification with peers. Useful when the node is running on a non-routable address or when the
+node does not have the capabilities to figure out its IP address.
+```toml
+external_address = ""
+```
+
+| Value type          | string                      |
+|:--------------------|:----------------------------|
+| **Possible values** | IP:port (`"1.2.3.4:26656"`) |
+|                     | `""`                        |
+
+The port has to point to the node's P2P port.
+
+Example with a node on a non-routable network:
+- Node has local IP address `10.10.10.10` and uses port `10000` for P2P communication, set in [`p2p.laddr`](#p2pladdr).
+- The network gateway has the public IP `1.2.3.4` and we want to use publicly open port `26656` on the IP address.
+- A redirection has to be set up from `1.2.3.4:26656` to `10.10.10.10:1000`
+- The `external_address` has to be set to `1.2.3.4:26656`.
+
+### p2p.seeds
+Comma-separated list of seed nodes that the node will try to connect.
+```toml
+seeds = ""
+```
+
+| Value type                        | string (comma-separated list)           |
+|:----------------------------------|:----------------------------------------|
+| **Possible values within commas** | nodeID@IP:port (`"abcd@1.2.3.4:26656"`) |
+|                                   | `""`                                    |
+
+Seed nodes will transmit a P2P address book for the node and then disconnect. No blocks or consensus communication takes
+place.
+
+Example:
+```toml
+seeds = "abcd@1.2.3.4:26656,deadbeef@5.6.7.8:10000"
+```
+
+### p2p.persistent_peers
+Comma-separated list of nodes to keep persistent connections to.
+```toml
+persistent_peers = ""
+```
+
+| Value type                        | string (comma-separated list)           |
+|:----------------------------------|:----------------------------------------|
+| **Possible values within commas** | nodeID@IP:port (`"abcd@1.2.3.4:26656"`) |
+|                                   | `""`                                    |
+
+Persistent peers do not count toward [`p2p.max_num_inbound_peers`](#p2pmax_num_inbound_peers) or
+[`p2p.max_num_outbound_peers`](#p2pmax_num_outbound_peers).
+
+Example:
+```toml
+persistent_peers = "fedcba@11.22.33.44:26656,beefdead@55.66.77.88:20000"
+```
+
+### p2p.addr_book_file
```

Comment:
Here we maybe can inform that the "in-memory" address book is periodically persisted to disk. Namely, the address book should not be empty when the node restarts.

Again in the discovery mechanism, we are first considering addresses in this file, not dialing seeds if it has good addresses.